python/cudf/cudf/_lib/scalar.pyx

```diff
@@ -107,18 +141,37 @@ cdef class Scalar:
         """
         Returns a host copy of the underlying device scalar.
         """
-        if pd.api.types.is_string_dtype(self.dtype):
-            return _get_py_string_from_string(self.c_value)
-        elif pd.api.types.is_numeric_dtype(self.dtype):
-            return _get_np_scalar_from_numeric(self.c_value)
-        elif pd.api.types.is_datetime64_dtype(self.dtype):
-            return _get_np_scalar_from_timestamp64(self.c_value)
-        elif pd.api.types.is_timedelta64_dtype(self.dtype):
-            return _get_np_scalar_from_timedelta64(self.c_value)
+        if self._host_value_current:
+            if self.is_valid():
+                return self._host_value
+            else:
+                return cudf.NA
         else:
-            raise ValueError(
-                "Could not convert cudf::scalar to a Python value"
-            )
+            result = self.get_device_value()
+            self._host_value = result
+            self._host_value_current = True
+            return result
+
+    def sync(self):
```

Comment:
Should we make this an internal method? `def _sync`

Given Scalars are going to be a public class I think we should hide the fact that we're using lazy semantics here.